pkg/kubelet/kubelet.go

```diff
@@ -2825,28 +2825,43 @@ func (kl *Kubelet) HandlePodSyncs(pods []*v1.Pod) {
 // canResizePod determines if the requested resize is currently feasible.
 // pod should hold the desired (pre-allocated) spec.
 // Returns true if the resize can proceed.
-func (kl *Kubelet) canResizePod(pod *v1.Pod) (bool, v1.PodResizeStatus, string) {
+// If the resize can proceed, the returned pod condition is nil. Otherwise,
+// the returned pod condition is of type PodResizePending with the reason
+// why the resize could not proceed.
+func (kl *Kubelet) canResizePod(pod *v1.Pod) (bool, *v1.PodCondition) {
 	if v1qos.GetPodQOS(pod) == v1.PodQOSGuaranteed && !utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScalingExclusiveCPUs) {
 		if utilfeature.DefaultFeatureGate.Enabled(features.CPUManager) {
 			if kl.containerManager.GetNodeConfig().CPUManagerPolicy == "static" {
 				msg := "Resize is infeasible for Guaranteed Pods alongside CPU Manager static policy"
 				klog.V(3).InfoS(msg, "pod", format.Pod(pod))
-				return false, v1.PodResizeStatusInfeasible, msg
+				kl.recorder.Eventf(pod, v1.EventTypeWarning, events.ResizeInfeasible, msg)
+				return false, &v1.PodCondition{
+					Type:    v1.PodResizePending,
+					Reason:  "Infeasible",
+					Status:  v1.ConditionTrue,
+					Message: msg,
+				}
 			}
 		}
 		if utilfeature.DefaultFeatureGate.Enabled(features.MemoryManager) {
 			if kl.containerManager.GetNodeConfig().MemoryManagerPolicy == "static" {
 				msg := "Resize is infeasible for Guaranteed Pods alongside Memory Manager static policy"
 				klog.V(3).InfoS(msg, "pod", format.Pod(pod))
-				return false, v1.PodResizeStatusInfeasible, msg
+				kl.recorder.Eventf(pod, v1.EventTypeWarning, events.ResizeInfeasible, msg)
+				return false, &v1.PodCondition{
+					Type:    v1.PodResizePending,
```

Comment:
was planning to add `observedGeneration` here after https://github.com/kubernetes/kubernetes/pull/130573 goes in (I can either update it in this PR or submit a follow up depending on which merges first)